### ISSUE TITLE
config help

### ISSUE THREAD

**Issue by neallawson** [onddo/dovecot-cookbook#12]
after reading your docs i could use some help, i would like to configure my dovecot install to use my openldap server for auth, and mail dir location

```json
        "dovecot": {
			"packages":["imap","lmtp","sieve"],
            "conf": {
                "conf_files_mode": "00640",
                "disable_plaintext_auth": false,
                "ldap": {
                    "auth_bind": true,
                    "base": "dc=somedomain,dc=com",
                    "uris": "ldaps://ldap.somedomain.com:636"
                },
                "log_path": "syslog",
                "mail_location": "sdbox:/srv/vmail/%u/Maildir",
                "mail_privileged_group": "mail",
                "opt_mail_location": "maildir:/srv/vmail/%u/Maildir",
                "syslog_facility": "mail"
            },
            "conf_files_group": "dovecot",
            "protocols": {
                "imap": {},
                "lda": {}
            },
            "services": {
                "auth": {
                    "listeners": [
                        {
                            "unix:/var/spool/postfix/private/auth": {
                                "group": "postfix",
                                "mode": "0666",
                                "user": "postfix"
                            }
                        }
                    ]
                }
            }
        },
```

any suggestions would be helpful

**Comment by neallawson** [comment]
i did look a those resources, and they were almost of no help, regardless of what i have tried, i can't get the cookbook to write out the ldap conf files, 

default['dovecot']['conf_files']['ldap'] = %w(
  dovecot-ldap.conf.ext
  conf.d/auth-ldap.conf.ext
)

**Comment by zuazo** [comment]
@neallawson, to write those files `node['dovecot']['auth']['ldap']` has to be a hash. But I think that's a bug, because it should check against `node['dovecot']['conf']['ldap']['auth_bind']` instead.

I will try to look in more detail the next weekend because I have to read the LDAP documentation more carefully. Until then, try setting these attributes:

```ruby
node.default['dovecot']['auth']['ldap']['passdb']['driver'] = 'ldap'
node.default['dovecot']['auth']['ldap']['passdb']['args'] = '/etc/dovecot/dovecot-ldap.conf.ext'
```

**Comment by neallawson** [comment]
i did try setting 
```ruby
node.default['dovecot']['auth']['ldap']['passdb']['driver'] = 'ldap'
node.default['dovecot']['auth']['ldap']['passdb']['args'] = '/etc/dovecot/dovecot-ldap.conf.ext'
```
however no joy..... Thanks for the suggestion.

**Comment by neallawson** [comment]
oddly node['dovecot']['conf']['ldap']['auth_bind'] was also set to "true"

**Comment by zuazo** [comment]
> i did try setting 
> ```ruby
> node.default['dovecot']['auth']['ldap']['passdb']['driver'] = 'ldap'
> node.default['dovecot']['auth']['ldap']['passdb']['args'] = '/etc/dovecot/dovecot-ldap.conf.ext'
> ```
> however no joy..... Thanks for the suggestion.

You mean that the LDAP config files are not created? Because I just tested it with test-kitchen and seems to be created :disappointed:

    root@ldap-ubuntu-1204:~# find /etc/dovecot/ -name '*ldap*' -ls
    1046746    8 -rw-r-----   1 root     dovecot      5242 Jan 13 04:52 /etc/dovecot/dovecot-ldap.conf.ext
    1046748    4 -rw-r--r--   1 root     dovecot       744 Jan 13 04:52 /etc/dovecot/conf.d/auth-ldap.conf.ext

The `auth_bind` option is ignored, so that is expected behavior.

**Comment by zuazo** [comment]
Hi @neallawson,

Sorry for the delay. I improved LDAP support in master. You can see [the recipe example used for test-kitchen](https://github.com/onddo/dovecot-cookbook/blob/master/test/cookbooks/dovecot_test/recipes/ldap.rb).

**Comment by zuazo** [comment]
LDAP fix released in [2.1.0](https://github.com/onddo/dovecot-cookbook/releases/tag/2.1.0).
